api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java

```diff
@@ -155,8 +159,13 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
     }
     DbUser user = userDao.findUserByUsername(userName);
```

Comment:
nit: you can use the new `Optional<DbUser> getByUsername(String username)` on `UserService` to avoid the null comparison.

Reply:
I made a good-faith attempt to use Optional patterns here, but I couldn't come up with a solution that wasn't more awkward than the current conditional logic. The challenge is that we want to *either* provide a non-null DbUser object if the flag allows it, *or* return false, which isn't easily expressed within chained optional lambdas.

There was a bug in my existing logic though, which I did fix (pushing new commit in a minute). So there's that...

Reply:
Yeah, it's hard to write awkward logic with Optionals, which is kind of the point.

FWIW you can do `DbUser user  = getByUsername(String username).orElseGet(() -> userService.createServiceAccountUser(userName));` to clean up one  conditional. Or better yet put a `getOrCreateServiceAccount(String username)` on the `UserService`.